llvm/lib/TargetParser/Host.cpp

```diff
@@ -707,9 +707,7 @@ static StringRef
 getIntelProcessorTypeAndSubtype(unsigned Family, unsigned Model,
                                 const unsigned *Features,
                                 unsigned *Type, unsigned *Subtype) {
-  auto testFeature = [&](unsigned F) {
-    return (Features[F / 32] & (1U << (F % 32))) != 0;
-  };
+#define testFeature(F) (Features[F / 32] & (1 << (F % 32))) != 0
```

Comment:
Should we move this define before the getIntel* definition and then undef it after we've finished with it? I'm not sure how clean it is to define this INSIDE the getIntel* function and then keep using it in later functions. I'm assuming we can't keep the lambda if its going to be used in c code in the future?

Reply:
Fixed. I moved the definition above `getIntelProcessorType*` and added an undef after `getAMDProcessorType*`. Right, the lambda would be cleaner, but lambdas are a C++ feature, which means we can't use it in a unified implementation between LLVM and compiler-rt.

Reply:
Move this define as well (and add undef)

Reply:
Good catch! Thanks! Updated.